contributors/design-proposals/local-storage-pv.md

```diff
@@ -0,0 +1,631 @@
+
+The DaemonSet for this example looks like:
+```
+
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: local-storage-provisioner
+  namespace: kube-system
+spec:
+  template:
+    metadata:
+      labels:
+        system: local-storage-provisioner
+    spec:
+      containers:
+      - name: provisioner
+        image: "gcr.io/google_containers/local-storage-provisioner:v1.0"
+        imagePullPolicy: Always
+        volumeMounts:
+        - name: vol1
+          mountPath: "/local-ssds"
+        - name: vol2
+          mountPath: "/local-hdds"
+        env:
+        - name: MY_NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+      volumes:
+      - name: vol1
+        hostPath:
+          path: "/mnt/ssds"
+      - name: vol2
+        hostPath:
+          path: "/mnt/hdds"
+      serviceAccount: local-storage-admin
+```
+
+##### Provisioner Boostrapper
+
+Manually setting up this DaemonSet spec can be tedious and it requires duplicate specification
+of the StorageClass -> directory mappings both in the ConfigMap and as hostPath volumes. To
+make it simpler and less error prone, a boostrapper application will be provided to generate
+and launch the provisioner DaemonSet based off of the ConfigMap.  It can also create a service
+account with all the required permissions.
+
+The boostrapper accepts the following optional arguments:
+
+* -image: Name of local volume provisioner image (default
+"quay.io/external_storage/local-volume-provisioner:latest")
+* -volume-config: Name of the local volume configuration configmap. The configmap must reside in the same
+namespace as the bootstrapper. (default "local-volume-default-config")
+* -serviceaccount: Name of the service accout for local volume provisioner (default "local-storage-admin")
+
+The boostrapper requires the following permissions:
+
+* Get/Create/Update ConfigMap
+* Create ServiceAccount
+* Create ClusterRoleBindings
+* Create DaemonSet
+
+Since the boostrapper generates the DaemonSet spec, the ConfigMap can be simplifed to just specify the
+host directories:
+
+```
+kind: ConfigMap
+metadata:
+  name: local-volume-config
+  namespace: kube-system
+data:
+  "local-fast": |
+    {
+      "hostDir": "/mnt/ssds",
+    }
+  "local-slow": |
+    {
+      "hostDir": "/mnt/hdds",
+    }
+```
+
+The boostrapper will update the ConfigMap with the generated `mountDir`.  It generates the `mountDir`
+by stripping off the initial "/" in `hostDir`, replacing the remaining "/" with "~", and adding the
+prefix path "/mnt/local-storage".
+
+In the above example, the generated `mountDir` is `/mnt/local-storage/mnt ~ssds` and
+`/mnt/local-storage/mnt~hdds`, respectively.
+
+
+#### Limitations
+
+This initial phase for Local PV support will provide the following:
+
+* Local directories to be specified as Local PVs with node affinity
+* Pod using a PVC that is bound to a Local PV will always be scheduled to that node
```

Comment:
what is the alternatives? why this is an limitation?

Reply:
Sorry i need to organize this section better.  I wanted to start off with the use cases that are supported in 1.7, and then follow up with the limitations in 1.7.  So this first section is describing what a user can do.

Reply:
Split up this section into deliverables and limitations.